app/src/main/java/io/homeassistant/companion/android/websocket/WebsocketManager.kt

```diff
@@ -0,0 +1,178 @@
+package io.homeassistant.companion.android.websocket
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.hardware.display.DisplayManager
+import android.os.Build
+import android.util.Log
+import android.view.Display
+import androidx.core.app.NotificationCompat
+import androidx.core.content.getSystemService
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.NetworkType
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.components.SingletonComponent
+import io.homeassistant.companion.android.common.BuildConfig
+import io.homeassistant.companion.android.common.R
+import io.homeassistant.companion.android.common.data.websocket.WebSocketRepository
+import io.homeassistant.companion.android.database.AppDatabase
+import io.homeassistant.companion.android.database.settings.WebsocketSetting
+import io.homeassistant.companion.android.notifications.MessagingManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.util.concurrent.TimeUnit
+
+@ExperimentalCoroutinesApi
+class WebsocketManager(
+    appContext: Context,
+    workerParams: WorkerParameters
+) : CoroutineWorker(appContext, workerParams) {
+
+    companion object {
+        private const val TAG = "WebSockManager"
+        private const val CHANNEL_ID = "Websocket"
+        private const val NOTIFICATION_ID = 65423
+        private val DEFAULT_WEBSOCKET_SETTING = if (BuildConfig.BUILD_TYPE == "minimal") WebsocketSetting.ALWAYS else WebsocketSetting.SCREEN_ON
```

Comment:
Should we make this always only if the build is `full`? I know we cant post notifications on the `quest` build yet but I think some notifications commands may work :)

Reply:
Fixed.